common/src/main/java/org/opensearch/ml/common/connector/ConnectorHttpClientConfig.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.connector;
+
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.common.io.stream.Writeable;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken;
+
+@Getter
+@EqualsAndHashCode
+public class ConnectorHttpClientConfig implements ToXContentObject, Writeable {
+
+    public static final String MAX_CONNECTION_FIELD = "max_connection";
+    public static final String CONNECTION_TIMEOUT_FIELD = "connection_timeout";
+    public static final String READ_TIMEOUT_FIELD = "read_timeout";
+
+    public static final Integer MAX_CONNECTION_DEFAULT_VALUE = Integer.valueOf(30);
+    public static final Integer CONNECTION_TIMEOUT_DEFAULT_VALUE = Integer.valueOf(20000);
+    public static final Integer READ_TIMEOUT_DEFAULT_VALUE = Integer.valueOf(20000);
```

Comment:
Waiting for 20s for user seems not a good idea, is this the suggested default value from OpenSearch?

Reply:
https://opensearch.org/docs/1.2/opensearch/rest-api/cluster-settings/#:~:text=Default%20is%2030%20seconds.



From here, it looks like the default value is 30 seconds. I'm planning to make this 30 seconds unless we have strong reasoning not to do this.

Reply:
The other way is to keep the default http timeout, we don't need to use 
```
this.readTimeout = Objects.requireNonNullElse(readTimeout, READ_TIMEOUT_DEFAULT_VALUE);
```
We just let `this.readTimeout = readTimeout` , if it's null, we don't need to set it, then client will use default timeout.